### PR TITLE
#100 prune solr conf files

### LINKED ISSUE #100: Review solr conf files
In GitLab by @bzk60 on Sep 13, 2018, 12:37

### PR DESCRIPTION
**Pull request by banukutlu**
Closes #100

### CODE REVIEW

**Comment by cdmo** [comment]
+9 −953

😎

File: solr/conf/protwords.txt [hunk 1]
```diff
@@ -16,6 +16,6 @@
 
 # Some non-words that normally won't be encountered,
 # just to test that they won't be stemmed.
-dontstems
-zwhacky
+#dontstems
+#zwhacky
```
Comment (cdmo) [inline solr/conf/protwords.txt]: Maybe we should leaves these in to test stemming? Or will we handle stemming testing later on?

Comment (banukutlu) [reply solr/conf/protwords.txt]: I was thinking to revisit these during MVP1 with this https://github.com/psu-libraries/psulib_blacklight/issues/139.

**Comment by banukutlu** [comment]
> In general I think there's a lot of default settings in solr config files that are there as a starting point and to test basic functionality. Like synonyms, spelling and stemming. Maybe we should hold off on deleting this stuff until we verify those things are working? Or maybe you did already?
> 
> I love when code is deleted, just want to make sure we'll be okay.

Most of the deleted files are either things that are not used in 7.4 anymore or things that won't use or  less likely to implement yet. I left stopwords, synonyms and protwords files which I thought will be the ones we will start tweaking soon. Re spellings.txt - this is only used with FileBasedSpellChecker. I think we have the default IndexBasedSpellChecker.

File: solr/conf/synonyms.txt [hunk 2]
```diff
@@ -12,12 +12,12 @@
```
[the remark this answers is not in the document]
Comment (banukutlu) [reply solr/conf/synonyms.txt]: you mean deleting the lines? just left them as examples but we can delete now or when we are tweaking that file.

[the remark this answers is not in the document]
Comment (cdmo) [reply solr/conf/synonyms.txt]: I approved the review, I think I get what you are doing. Leaving comments is sort of like leaving breadcrumbs to us to take care of this as I see it. Right?

[the remark this answers is not in the document]
Comment (banukutlu) [reply solr/conf/synonyms.txt]: yes exactly :)

File: solr/conf/xslt/luke.xsl [hunk 3]
```diff
@@ -1,337 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (banukutlu) [reply solr/conf/xslt/luke.xsl]: luke.xsl is under xslt folder and my understanding is that which would be used for mostly on-the-fly XSLT transformations when indexing XML files.  

If we want to set up Luke,  https://wiki.apache.org/solr/LukeRequestHandler - just adding 
`<requestHandler name="/admin/luke" class="solr.admin.LukeRequestHandler" />` to solrconfig does the trick. (just tried on my local and seems to be working.  http://localhost:8983/solr/blacklight-core/admin/luke.)

[the remark this answers is not in the document]
Comment (cdmo) [reply solr/conf/xslt/luke.xsl]: Cool 😎

**Pull request merged**